### ISSUE TITLE
Server side validation

### ISSUE THREAD

**Issue by madmax3365** [marmelab/react-admin#1809]
Greetings. Is there any way to perform server side form validation?

**Comment by djhi** [comment]
Hi, and thanks for your question. As explained in the [react-admin contributing guide](https://github.com/marmelab/react-admin/blob/master/CONTRIBUTING.md), the right place to ask a "How To" question, get usage advice, or troubleshoot your own code, is [StackOverFlow](http://stackoverflow.com/questions/tagged/react-admin).

This makes your question easy to find by the core team, and the developer community. Unlike Github, StackOverFlow has great SEO, gamification, voting, and reputation. That's why we chose it, and decided to keep GitHub issues only for bugs and feature requests.

So I'm closing this issue, and inviting you to ask your question at:

http://stackoverflow.com/questions/tagged/react-admin

And once you get a response, please continue to hang out on the admin-on-rest channel in StackOverflow. That way, you can help newcomers and share your expertise!

**Comment by christiaanwesterbeek** [comment]
Related:

https://stackoverflow.com/questions/44190490/show-server-side-validation-errors-after-failed-form-submit
https://stackoverflow.com/questions/50237163/how-can-i-show-customized-error-messaged-from-server-side-validation-in-react-ad

https://github.com/marmelab/react-admin/pull/871
https://github.com/marmelab/react-admin/pull/1354

https://marmelab.com/react-admin/CreateEdit.html#validation

**Comment by christiaanwesterbeek** [comment]
@djhi 

In your last sentence, there's still a reference to admin-on-rest.

**Comment by djhi** [comment]
Thx updated :)

**Comment by nicolaskern** [comment]
Hi,

When throwing a SubmissionError exception during handleSubmit of SimpleForm component, the error keeps being unhandled with the message "SubmissionError: Submit Validation Failed".
I had the same error with admin-on-rest : https://github.com/marmelab/admin-on-rest/issues/24

Thanks,
Nicolas

**Comment by trsteel88** [comment]
Anyone able to help with this? I am throwing the error from my dataprovider but the errors don't map to the fields. Just get "Submit Validation Failed" in the footer.

```
return Promise.reject(new SubmissionError({name: ['test']}));
```

**Comment by llioor** [comment]
Because it appear on Google I will add my solution:
stackoverflow.com/a/54697878/2862728

**Comment by Benew** [comment]
Did this break in v3?

**Comment by bucuri** [comment]
> Did this break in v3?

In V3 it changed because it uses react-final-form but I think it was also broken. It is fixed starting with 3.12.0 (I just looked at the code).
